Examples/Io/Root/src/RootNTupleReader.cpp

```diff
@@ -0,0 +1,417 @@
+
+    cov(Acts::BoundIndices::eBoundLoc1, Acts::BoundIndices::eBoundLoc0) =
+        (*m_branches.m_track_cov_d0z0)[i];
+    cov(Acts::BoundIndices::eBoundPhi, Acts::BoundIndices::eBoundLoc0) =
+        (*m_branches.m_track_cov_d0phi)[i];
+    cov(Acts::BoundIndices::eBoundTheta, Acts::BoundIndices::eBoundLoc0) =
+        (*m_branches.m_track_cov_d0theta)[i];
+    cov(Acts::BoundIndices::eBoundQOverP, Acts::BoundIndices::eBoundLoc0) =
+        (*m_branches.m_track_cov_d0qOverP)[i];
+    cov(Acts::BoundIndices::eBoundPhi, Acts::BoundIndices::eBoundLoc1) =
+        (*m_branches.m_track_cov_z0phi)[i];
+    cov(Acts::BoundIndices::eBoundTheta, Acts::BoundIndices::eBoundLoc1) =
+        (*m_branches.m_track_cov_z0theta)[i];
+    cov(Acts::BoundIndices::eBoundQOverP, Acts::BoundIndices::eBoundLoc1) =
+        (*m_branches.m_track_cov_z0qOverP)[i];
+    cov(Acts::BoundIndices::eBoundTheta, Acts::BoundIndices::eBoundPhi) =
+        (*m_branches.m_track_cov_phitheta)[i];
+    cov(Acts::BoundIndices::eBoundQOverP, Acts::BoundIndices::eBoundPhi) =
+        (*m_branches.m_track_cov_phiqOverP)[i];
+    cov(Acts::BoundIndices::eBoundQOverP, Acts::BoundIndices::eBoundTheta) =
+        (*m_branches.m_track_cov_tehtaqOverP)[i];
+
+    Acts::BoundTrackParameters tc(surface, params, q, cov);
+    trackContainer.push_back(tc);
+  }
+
+  std::vector<Acts::Vector4> truthVertexContainer;
+  for (unsigned int i = 0; i < nTruthVtx; i++) {
+    Acts::Vector4 vtx(
+        (*m_branches.m_truthvertex_x)[i], (*m_branches.m_truthvertex_y)[i],
+        (*m_branches.m_truthvertex_z)[i], (*m_branches.m_truthvertex_t)[i]);
+    truthVertexContainer.push_back(vtx);
+  }
+  std::vector<Acts::Vector4> recoVertexContainer;
+  for (unsigned int i = 0; i < nRecoVtx; i++) {
+    Acts::Vector4 vtx((*m_branches.m_recovertex_x)[i],
+                      (*m_branches.m_recovertex_y)[i],
+                      (*m_branches.m_recovertex_z)[i], 0);
+    recoVertexContainer.push_back(vtx);
+  }
+
+  Acts::Vertex<Acts::BoundTrackParameters> beamspotConstraint;
+  Acts::Vector3 beamspotPos;
+  Acts::SymMatrix3 beamspotCov;
+
+  beamspotPos << m_branches.m_beamspot_x, m_branches.m_beamspot_y,
+      m_branches.m_beamspot_z;
+  beamspotCov << m_branches.m_beamspot_sigX * m_branches.m_beamspot_sigX, 0, 0,
+      0, m_branches.m_beamspot_sigY * m_branches.m_beamspot_sigY, 0, 0, 0,
+      m_branches.m_beamspot_sigZ * m_branches.m_beamspot_sigZ;
+
+  beamspotConstraint.setPosition(beamspotPos);
+  beamspotConstraint.setCovariance(beamspotCov);
+
+  context.eventStore.add(m_cfg.outputTrackParameters,
+                         std::move(trackContainer));
+  context.eventStore.add(m_cfg.outputTruthVtxParameters,
+                         std::move(truthVertexContainer));
+  context.eventStore.add(m_cfg.outputRecoVtxParameters,
+                         std::move(recoVertexContainer));
+  context.eventStore.add(m_cfg.outputBranchPointerWrapper, &m_branches);
```

Comment:
We seem to add a pointer here to the whiteboard... this is problematic from a livetime perspective, right?

Actually I do not understand why we write this to the whiteborad at all, since we take the things from `m_branches` before and add them to the other containers...

Reply:
true not sure what this was used for. I removed it